modules/gdscript/gdscript_parser.h

```diff
@@ -152,24 +152,34 @@ class GDScriptParser {
 		_FORCE_INLINE_ String to_string_strict() const { return is_hard_type() ? to_string() : "Variant"; }
 		PropertyInfo to_property_info(const String &p_name) const;
 
-		_FORCE_INLINE_ void set_container_element_type(const DataType &p_type) {
-			container_element_type = memnew(DataType(p_type));
+		_FORCE_INLINE_ int container_element_type_size() const {
+			return container_element_types.size();
 		}
 
-		_FORCE_INLINE_ DataType get_container_element_type() const {
-			ERR_FAIL_NULL_V(container_element_type, DataType());
-			return *container_element_type;
+		_FORCE_INLINE_ void set_container_element_type(int p_index, const DataType &p_type) {
+			while (container_element_type_size() <= p_index) {
+				container_element_types.push_back(nullptr);
+			}
+			container_element_types.set(p_index, memnew(DataType(p_type)));
 		}
 
-		_FORCE_INLINE_ bool has_container_element_type() const {
-			return container_element_type != nullptr;
+		_FORCE_INLINE_ DataType get_container_element_type(int p_index) const {
+			ERR_FAIL_INDEX_V(p_index, container_element_type_size(), DataType());
+			ERR_FAIL_NULL_V(container_element_types[p_index], DataType());
+			return *container_element_types[p_index];
 		}
 
-		_FORCE_INLINE_ void unset_container_element_type() {
-			if (container_element_type) {
-				memdelete(container_element_type);
-			};
-			container_element_type = nullptr;
+		_FORCE_INLINE_ bool has_container_element_type(int p_index) const {
+			return container_element_type_size() > p_index && container_element_types[p_index] != nullptr;
+		}
+
+		_FORCE_INLINE_ void unset_container_element_types() {
+			for (int i = 0; i < container_element_type_size(); i++) {
+				if (container_element_types[i]) {
```

Comment:
```suggestion
				if (container_element_types[i] != nullptr) {
```

Just for readability and consistency with `has_container_element_type()`.